pylint/checkers/utils.py

```diff
@@ -434,8 +435,20 @@ def overrides_a_method(class_node: nodes.ClassDef, name: str) -> bool:
     return False
 
 
-def check_messages(*messages: str) -> Callable:
-    """Decorator to store messages that are handled by a checker method."""
+def only_required_for_messages(
+    *messages: str,
+) -> Callable[
+    [Callable[[Any, nodes.NodeNG], None]], Callable[[Any, nodes.NodeNG], None]
+]:
```

Comment:
@DanielNoord the typing gets really longish here. Are we sure that we want to keep it?

Also, I initially tried to put ``BaseChecker`` as first argument instead of ``Any``, but even though I think this should be correct ``mypy`` failed, because we don't have ``BaseChecker`` instances but instances of subclasses.

Reply:
Maybe we need to do something like this:
```
from typing import TypeVar
C = TypeVar('C', bound=BaseChecker)
```

```
    [Callable[[C, nodes.NodeNG], None]], Callable[[C, nodes.NodeNG], None]
```
Based on https://mypy.readthedocs.io/en/stable/generics.html (The Friend / SuperFriend example)

Reply:
Thanks, that worked!